src/hssm/hssm.py

```diff
@@ -678,7 +804,25 @@ def sample_prior_predictive(
             ``InferenceData`` object with the groups ``prior``, ``prior_predictive`` and
             ``observed_data``.
         """
-        return self.model.prior_predictive(draws, var_names, omit_offsets, random_seed)
+        prior_predictive = self.model.prior_predictive(
+            draws, var_names, omit_offsets, random_seed
+        )
+
+        prior_predictive.add_groups(posterior=prior_predictive.prior)
+        self.model.predict(prior_predictive, kind="mean", inplace=True)
+
+        # clean
+        prior_predictive.prior = prior_predictive.posterior
+        del prior_predictive.posterior
+
+        if self._inference_obj is None:
+            self._inference_obj = prior_predictive
+        else:
+            self._inference_obj.extend(prior_predictive)
+
+        # clean up `rt,response_mean` to `v`
+        self._drop_parent_str_from_idata(self._inference_obj)
```

Comment:
The type checker is a bit stupid here since the `if` block above basically indicates `self._inference_obj` is not None at this point. Maybe the easiest way to do this is to move this line above Line 818 and change the argument to `prior_predictive`?

Reply:
ok worked out the type checking stuff